### PR TITLE
Feat/wallet pubkey service

### PR DESCRIPTION
**Pull request by plondon**
## Description (optional)
TODO

## Testing Steps (optional)
Detail the steps required for the reviewer(s) to verify and test these changes.

### CODE REVIEW

**Comment by kieran-bc** [comment]
Hey Phil, can you give a bit of a description of the core classes involved here and what changes to those classes have caused the need to touch all the other files.  I'll take a stab at reading the code regardless.

File: packages/blockchain-wallet-v4-frontend/src/data/components/request/sagas.ts [hunk 1]
```diff
@@ -23,33 +26,35 @@ export default ({ api, coreSagas, networks }: { api: APIType; coreSagas: any; ne
...
       let address
-      const extras: RequestExtrasType = {}
+      let extras: RequestExtrasType = {}
       const { account } = action.payload
+      const { coinfig } = window.coins[account.coin]
 
+      const subscriptions = S.getSubscriptions(yield select())
+      const isSubscribed = subscriptions.data.currencies.some((c) => c.ticker === account.baseCoin)
+      if (!isSubscribed) {
+        yield put(actions.core.data.coins.subscribe(account))
```
Comment (plondon) [inline packages/blockchain-wallet-v4-frontend/src/data/components/request/sagas.ts:40]: subscribe to coin

File: packages/blockchain-wallet-v4-frontend/src/data/components/sendCrypto/sagas.ts [hunk 2]
```diff
@@ -365,6 +380,14 @@ export default ({ api }: { api: APIType }) => {
     const isDynamicSelfCustody = window.coins[coin].coinfig.products.includes('DynamicSelfCustody')
 
     try {
+      if (
+        getCoinNetworksAndTypes().networks[coin] &&
+        getCoinNetworksAndTypes().networks[coin].type === 'EVM'
+      ) {
+        yield put(A.validateAddressSuccess(ethers.utils.isAddress(address)))
+        return
+      }
```
Comment (plondon) [inline packages/blockchain-wallet-v4-frontend/src/data/components/sendCrypto/sagas.ts:389]: can remove this now

File: packages/blockchain-wallet-v4-frontend/src/scenes/CoinPage/components/CoinPage/hooks/useActivityFeed/useActivityFeed.tsx [hunk 3]
```diff
@@ -123,7 +123,7 @@ const useActivityFeed: ActivityFeedHook = ({ coin }) => {
             />
           </Button>
 
-          {!isLoadingAddressData && (
+          {!isLoadingAddressData && filterOptions.length > 1 && (
```
Comment (plondon) [inline packages/blockchain-wallet-v4-frontend/src/scenes/CoinPage/components/CoinPage/hooks/useActivityFeed/useActivityFeed.tsx:126]: not sure what this is

File: packages/blockchain-wallet-v4/src/redux/data/coins/sagas.ts [hunk 4]
```diff
@@ -137,60 +117,403 @@ export default ({ api }: { api: APIType }) => {
...
+      //     description: prop('description', tx),
+      //     exchange_rate_then: fiatCurrencySymbol + priceAtTime.toFixed(2),
+      //     // @ts-ignore
+      //     hash: prop('hash', tx),
+      //     // @ts-ignore
+      //     time: tx.time,
+      //     type: txType,
+      //     value_now: `${fiatCurrencySymbol}${negativeSignOrEmpty}${valueNow}`,
+      //     value_then: `${fiatCurrencySymbol}${negativeSignOrEmpty}${valueThen}`
+      // }
```
Comment (plondon) [inline packages/blockchain-wallet-v4/src/redux/data/coins/sagas.ts:178]: delete

File: packages/blockchain-wallet-v4/src/redux/data/coins/selectors.ts [hunk 5]
```diff
@@ -90,6 +85,91 @@ export const getTransactions = (coin: string, state: RootState) => {
...
+          return balance
+        }
+
+        return lift(transform)(unifiedBalancesR)
+      }
+    )(state)
+  }
+}
+
+export const getCoinNetworksAndTypes = () => ({
```
Comment (plondon) [inline packages/blockchain-wallet-v4/src/redux/data/coins/selectors.ts:122]: delete this, get from BE eventually

File: packages/blockchain-wallet-v4/src/redux/data/self-custody/sagas.ts [hunk 6]
```diff
@@ -6,6 +6,17 @@ import * as selectors from '../../selectors'
 
 const taskToPromise = (t) => new Promise((resolve, reject) => t.fork(reject, resolve))
 
+const getPath = (coin) => {
```
Comment (plondon) [inline packages/blockchain-wallet-v4/src/redux/data/self-custody/sagas.ts:9]: leave a comment here

**Pull request closed without merging**